### PR TITLE
quest list: stop chat input from closing when pressing enter

### LINKED ISSUE #11078: Quest Filter after Pressing Enter.
Update suggestion for the quest search:
After pressing enter when using search, only show the search results, instead of the whole quest list.
...and add a "show all" button. 

Additional:
Quest Grouping by difficulty:
Easy
Medium
Hard
Master
Grandmaster

Quest Grouping by length:
Very Short
Short
Medium
Long
Very Long

Quest requirements upon hovering quest title:
Toogleable options: (show all/show skills/show quests/show questpoints.)

Quest filtering by availability and new color grouping:

Dark Red: Unstarted, Unable to Start, Unable to Complete.
Light Red: Unstarted, Able to Start, Unable to Complete.
Orange: Started, Unable to Complete.
Cyan/Light Green: Started, Able to Complete.
Green: Completed.

Quest reordering by:
Name
Difficulty
Length
Availability
F2p/P2p

### PR DESCRIPTION
**Pull request by DSerp**
Closes #11078 

Stops the quest filter input from closing when pressing enter

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsPlugin.java [hunk 1]
```diff
@@ -408,6 +408,7 @@ public void onMenuOptionClicked(MenuOptionClicked event)
 				.addCharValidator(FILTERED_CHARS)
 				.value(initialValue)
 				.onDone((newValue) ->
+				{
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/banktags/BankTagsPlugin.java:411]: Why is this file changed?

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java [hunk 2]
```diff
@@ -269,6 +269,7 @@ private void handleDeposit(MenuOptionClicked event, Boolean inventory)
 		chatboxPanelManager.openTextInput((inventory ? "Inventory " : "Equipment ") + " tags:")
 			.addCharValidator(FILTERED_CHARS)
 			.onDone((newTags) ->
+			{
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java:272]: Why is this block changed?

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java [hunk 3]
```diff
@@ -290,15 +292,18 @@ private void handleNewTab(ScriptEvent event)
 			case NewTab.NEW_TAB:
 				chatboxPanelManager.openTextInput("Tag name")
 					.addCharValidator(FILTERED_CHARS)
-					.onDone((tagName) -> clientThread.invoke(() ->
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java:293]: Why is this block changed?

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java [hunk 4]
```diff
@@ -678,10 +683,10 @@ public void handleClick(MenuOptionClicked event)
 		}
 
 		if ((event.getWidgetId() == WidgetInfo.BANK_ITEM_CONTAINER.getId()
-				|| event.getWidgetId() == WidgetInfo.BANK_INVENTORY_ITEMS_CONTAINER.getId())
+			|| event.getWidgetId() == WidgetInfo.BANK_INVENTORY_ITEMS_CONTAINER.getId())
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java:686]: Revert this tab change. It is used to show nested boolean operations.
```suggestion
				|| event.getWidgetId() == WidgetInfo.BANK_INVENTORY_ITEMS_CONTAINER.getId())
```

File: runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java [hunk 5]
```diff
@@ -678,10 +683,10 @@ public void handleClick(MenuOptionClicked event)
...
 			&& event.getMenuAction() == MenuAction.CC_OP_LOW_PRIORITY
 			&& (event.getMenuOption().equalsIgnoreCase("withdraw-x")
-				|| event.getMenuOption().equalsIgnoreCase("deposit-x")))
+			|| event.getMenuOption().equalsIgnoreCase("deposit-x")))
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java:689]: Revert this tab change. It is used to show nested boolean operations.
```suggestion
				|| event.getMenuOption().equalsIgnoreCase("deposit-x")))
```

**Comment by DSerp** [comment]
> The changes to ChatboxTextInput should be in a separate commit from the `onDone` addition to the quest list plugin.
> 
> Also, all of these block wrappings should be removed as they do not change functionality and muddy up the diff significantly.

It's now in two separate commits.

Those block wrappings did have a purpose. It was to stop the lambda expression from being ambiguous with either the Consumer or the Predicate of the onDone methods. I have now changed it to a cast fix so it may be cleaner now.

File: runelite-client/src/main/java/net/runelite/client/game/chatbox/ChatboxTextInput.java [hunk 6]
```diff
@@ -235,6 +235,16 @@ public ChatboxTextInput onClose(Runnable onClose)
...
+	{
+		this.onDone = (s) ->
+		{
+			onDone.accept(s);
+			return true;
+		};
+		return this;
+	}
+
+	public ChatboxTextInput onDone(Predicate<String> onDone)
```
Comment (abextm) [inline runelite-client/src/main/java/net/runelite/client/game/chatbox/ChatboxTextInput.java:247]: this should get some documentation on what the return value is for

**Comment by DSerp** [comment]
I added the same logic to the other filters that you recommended. I as well added a comment to onDone.
